### PR TITLE
Add ml_collection to VAE

### PR DESCRIPTION
**Pull request by vballoli**
This PR adds ml_collection to the VAE example in reference to this [issue](https://github.com/google/flax/issues/440)

### CODE REVIEW

**Comment by mohitreddy1996** [comment]
Hi, 

Thank you for the PR. 

I would prefer using `config_flags.DEFINE_config_file` instead of `config_flags.DEFINE_config_dict` as the former accepts a file path to a python file which has `get_config()` method which returns an instance of `ConfigDict`. This allows having multiple python files defining multiple configurations (say depending on the hardware on which you train your model or platform (cloud, locally)) and makes defining hyper parameters more pythonic. 

See https://github.com/google/flax/pull/463 for reference.

**Comment by vballoli** [comment]
Ah, I did think of that initially, but since the number of values weren't high, I decided to go with the dict implementation. I'll revert to the file and fix the PR. Thank you for the feedback !

File: examples/vae/configs/default.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+"""Default Hyperparameter configuration."""
+
+import ml_collections
+
+
+def get_config():
+  """Get the default hyperparameter configuration."""
+  config = ml_collections.ConfigDict()
+
+  config.learning_rate = 0.1
```
Comment (mohitreddy1996) [inline examples/vae/configs/default.py:24]: Any reason to have the values here different than flag default values?

File: examples/vae/requirements.txt [hunk 2]
```diff
@@ -1,4 +1,6 @@
 jax
+jaxlib
 flax
 tensorflow
 pillow
+git+https://github.com/google/ml_collections
```
Comment (mohitreddy1996) [inline examples/vae/requirements.txt:6]: ml-collections is available as a pip package :) 

https://pypi.org/project/ml-collections/

File: examples/vae/train.py [hunk 3]
```diff
@@ -12,45 +12,32 @@
...
 import jax
 from jax import random
 
-from flax import nn
-from flax import optim
-
 import tensorflow as tf
 import tensorflow_datasets as tfds
 
+from flax import nn
```
Comment (mohitreddy1996) [inline examples/vae/train.py:31]: I would recommend reverting this change as it is not related to the problem being solved. 

There are such linter occurrences in almost all the examples, but it would be nice to have a wider fix or follow up PR for the same.

File: examples/vae/README.md [hunk 4]
```diff
@@ -5,7 +5,13 @@ This code follows [pytorch/examples/vae](https://github.com/pytorch/examples/blo
 
 ```bash
 pip install -r requirements.txt
-python main.py
+python train.py --config.batch_size=batch_size --config.learning_rate=lr
```
Comment (mohitreddy1996) [inline examples/vae/README.md:8]: Actually it would be nice to have the README consistent across examples. Could you rephrase to have something similar to: https://github.com/google/flax/blob/master/examples/mnist/README.md

**Review by mohitreddy1996 — CHANGES_REQUESTED** [review]

**Comment by mohitreddy1996** [comment]
ping :)

**Comment by vballoli** [comment]
Sorry about the delay, will fix the issues in some time !!!

**Comment by avital** [comment]
Hi @vballoli -- this PR has been open for a while, I'll close it for now but feel free to re-open if you'd like to see it through!
